File: web/lib/models/rating.dart
import 'package:json_annotation/json_annotation.dart';
import 'bioskopina.dart';
import 'user.dart';

part 'rating.g.dart';

@JsonSerializable()
class Rating {
  int? id;
  int? userId;
  int? movieId;
  int? ratingValue;
  String? reviewText;
  DateTime? dateAdded;
  Bioskopina? movie;
  User? user;

  Rating(
      this.id,
      this.userId,
      this.movieId,
      this.ratingValue,
      this.reviewText,
      this.dateAdded,
      this.movie,
      this.user,
      );

  factory Rating.fromJson(Map<String, dynamic> json) => _$RatingFromJson(json);

  Map<String, dynamic> toJson() => _$RatingToJson(this);
}

File: backend/Bioskopina.Services/Database/Rating.cs
using System;
using System.Collections.Generic;

namespace Bioskopina.Services.Database;

public partial class Rating
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public int MovieId { get; set; }

    public int? RatingValue { get; set; }

    public string ReviewText { get; set; } = null!;

    public DateTime DateAdded { get; set; }

    public virtual Bioskopina Movie{ get; set; } = null!;

    public virtual User User { get; set; } = null!;
}
